### PR TITLE
Add missing help text for some CMS options

### PR DESCRIPTION
**Pull request by richsalz**
I read the code for apps/cms.c, and traced through the underlying ```CMS_xxx``` API calls.
Unfortunately many of those functions are not documented.
I tried to make educated guesses as to what is going on.

This could be backported.

### CODE REVIEW

**Comment by richsalz** [comment]
Added "Fixes: #9952" to the commit.

**Comment by richsalz** [comment]
Updated commit pushed based on information from the cms.pod page.

File: apps/cms.c [hunk 1]
```diff
@@ -104,20 +104,28 @@ const OPTIONS cms_options[] = {
...
+    {"digest_create", OPT_DIGEST_CREATE, '-',
+        "Create a CMS \"DigestedData\" object"},
+    {"compress", OPT_COMPRESS, '-', "Create a CMS \"CompressedData\" object"},
+    {"uncompress", OPT_UNCOMPRESS, '-', "Uncompress a CMS \"CompressedData\" object"},
+    {"EncryptedData_decrypt", OPT_ED_DECRYPT, '-',
+        "Decrypt CMS \"EncryptedData\" object using symmetric key"},
+    {"EncryptedData_encrypt", OPT_ED_ENCRYPT, '-',
+        "Create CMS \"EncryptedData\" object using symmetric key"},
+    {"debug_decrypt", OPT_DEBUG_DECRYPT, '-',
+        "Ignore errors when decryption recipients (see documentation)"},
```
Comment (t8m) [inline apps/cms.c]: decrypting

Comment (richsalz) [reply apps/cms.c]: thanks, fixed.

Comment (t8m) [reply apps/cms.c]: unfortunately with decripting and not decrypting

Comment (richsalz) [reply apps/cms.c]: gah.  udated commit pushed.

File: apps/cms.c [hunk 2]
```diff
@@ -104,20 +104,28 @@ const OPTIONS cms_options[] = {
     {"resign", OPT_RESIGN, '-', "Resign a signed message"},
     {"cades", OPT_CADES, '-', "Include signer certificate digest"},
     {"verify", OPT_VERIFY, '-', "Verify signed message"},
-    {"verify_retcode", OPT_VERIFY_RETCODE, '-'},
-    {"verify_receipt", OPT_VERIFY_RECEIPT, '<'},
+    {"verify_retcode", OPT_VERIFY_RETCODE, '-',
+        "Exit non-zero on verification failure"},
+    {"verify_receipt", OPT_VERIFY_RECEIPT, '<',
+        "Exit with failure if receipt signatures do not verify"},
```
Comment (beldmit) [inline apps/cms.c]: Not sure. If I understand correctly, it just invokes the SMIME_VERIFY_RECEIPT

File: apps/cms.c [hunk 3]
```diff
@@ -104,20 +104,28 @@ const OPTIONS cms_options[] = {
...
+    {"digest_create", OPT_DIGEST_CREATE, '-',
+        "Create a CMS \"DigestedData\" object"},
+    {"compress", OPT_COMPRESS, '-', "Create a CMS \"CompressedData\" object"},
+    {"uncompress", OPT_UNCOMPRESS, '-', "Uncompress a CMS \"CompressedData\" object"},
+    {"EncryptedData_decrypt", OPT_ED_DECRYPT, '-',
+        "Decrypt CMS \"EncryptedData\" object using symmetric key"},
+    {"EncryptedData_encrypt", OPT_ED_ENCRYPT, '-',
+        "Create CMS \"EncryptedData\" object using symmetric key"},
+    {"debug_decrypt", OPT_DEBUG_DECRYPT, '-',
+        "Ignore errors when decrypting recipients (see documentation)"},
```
Comment (beldmit) [inline apps/cms.c]: It seems to contradict to the documentation:

disable the MMA attack protection and return an error if no recipient can be found

File: apps/cms.c [hunk 4]
```diff
@@ -129,16 +137,20 @@ const OPTIONS cms_options[] = {
...
     {"noindef", OPT_NOINDEF, '-', "Disable CMS streaming"},
     {"crlfeol", OPT_CRLFEOL, '-', "Use CRLF as EOL termination instead of CR only" },
     {"noout", OPT_NOOUT, '-', "For the -cmsout operation do not output the parsed CMS structure"},
     {"receipt_request_print", OPT_RR_PRINT, '-', "Print CMS Receipt Request" },
-    {"receipt_request_all", OPT_RR_ALL, '-'},
-    {"receipt_request_first", OPT_RR_FIRST, '-'},
+    {"receipt_request_all", OPT_RR_ALL, '-',
+        "When signing, create receipt request for all recipients"},
+    {"receipt_request_first", OPT_RR_FIRST, '-',
+        "When signing, create receipt request for first recipient"},
```
Comment (beldmit) [inline apps/cms.c]: Nit: missing 'the'?

File: apps/cms.c [hunk 5]
```diff
@@ -151,10 +163,13 @@ const OPTIONS cms_options[] = {
      "Supply or override content for detached signature"},
     {"print", OPT_PRINT, '-',
      "For the -cmsout operation print out all fields of the CMS structure"},
-    {"secretkey", OPT_SECRETKEY, 's'},
-    {"secretkeyid", OPT_SECRETKEYID, 's'},
-    {"pwri_password", OPT_PWRI_PASSWORD, 's'},
-    {"econtent_type", OPT_ECONTENT_TYPE, 's'},
+    {"secretkey", OPT_SECRETKEY, 's',
+        "Use specified key to decrypt/encrypt recipients or content"},
```
Comment (beldmit) [inline apps/cms.c]: Suggest to add 'hex-encoded' for clarity.

**Comment by richsalz** [comment]
Thanks @beldmit, updated commit pushed to address your feedback.

**Comment by beldmit** [comment]
Merged. Many thanks!

**Comment by richsalz** [comment]
thanks for the helpful reviews!

**Pull request closed without merging**